websrc/source/documents/community/CONTRIBUTING.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: "documents"
+page_title: "Contiv Community"
+sidebar_current: "community"
+description: |-
+  Contributing to Contiv
+---
+# Contributing to Contiv
+
+Submitting code changes is only one of the many ways to contribute.
+Reporting issues, proposing documentation and design changes,
+discussing use cases and proposing integration with other software
+from the ecosystem are more ways to contribute to Contiv.
+Contributors can also become maintainers or committers.
+
+All contributions are welcome, no matter how small or how big they are.
+
+Various repos may have more specific requirements for contribution - check out CONTRIBUTING.md
+files in respective repos.
+
+## Reporting Issues
+You can report an issue in a specific repo by 
+clicking `New Issue` in the respective repo.
+Issues should also be opened for feature requests.
+
+## Submitting pull requests for code or documentation changes
+Changes can be proposed by sending a pull request (PR). One of the committers
+will review the changes and provide feedback. The pull request will be merged
+into the master branch after discussion.
+
+Please make sure that the unit tests and the system tests pass before
+submitting the PR, as appropriate.
+Please keep in mind some changes might not be merged if committers
+decide they can't be merged.
+
+
+## Submitting proposals for major changes
+Please include `Proposal: ...` in the title of the issue if you wish to
+do significant refactoring of the code, to propose a new component or to
+introduce a major change. Marking the issue as a proposal will ensure that
+more people provide feedback as early as possible.
+
+Significant code changes submitted without an accompanying proposal might
+be rejected and not merged. Such significant submissions without a proposal are
+discouraged to avoid wasting time.
+
+
+After design discussions:
+- Fork the repository to your own public repository
+- Make the changes in your repository in a new branch
+- Add unit and system test cases for your code, as appropriate
+- Make sure existing tests and newly added tests pass, as appropriate
+- Rebase your branch on top of the latest master branch
+- Re-run the unit and system tests
+- Submit a pull request with the code changes
+- A discussion may take place on your pull request
+- Discussing while writing the code is also recommended
+- Requested changes should be made to the same branch on your fork
+- Changes should be force pushed to the same branch of the pull request
+- The unit and system tests need to be run again after committers `LGTM` the change
```

Comment:
This is not necessarily true.  Code which has tests in progress or tests which already passed could be approved by a maintainer and subsequently merged without any further action by the submitter.  Since we gate merges on CI being in a passing state, I don't know if we need to call this out specifically.

Reply:
dseerv: That will vary a bit based on committer discretion.  Just because a pull request passes at one point of time does not mean it will pass at the point its eventually merged.  Its why we have committers code reviewing, so they can make an intelligent assessment of risk.  I think the over all concern can be worked out with a few 'mays' ;)

Reply:
@hagbard5235 makes sense, thanks